ui/src/app.tsx

```diff
@@ -23,6 +27,59 @@ export function App() {
         history
     };
 
+    // Add a small delay to ensure all resources are properly loaded
+    useEffect(() => {
+        const timer = setTimeout(() => {
+            try {
+                // Any initialization logic that might fail can go here
+                setIsLoading(false);
+            } catch (err) {
+                setError(err instanceof Error ? err : new Error('An unexpected error occurred'));
+                setIsLoading(false);
+            }
+        }, 300);
+
+        return () => clearTimeout(timer);
+    }, []);
+
+    // Global error boundary
+    useEffect(() => {
+        const handleError = (event: ErrorEvent) => {
+            event.preventDefault();
+            setError(event.error || new Error(event.message));
+        };
+
+        window.addEventListener('error', handleError);
+
+        return () => {
+            window.removeEventListener('error', handleError);
+        };
+    }, []);
+
+    // Handle any errors during initialization
+    if (error) {
```

Comment:
This seems to be duplicating the functionality of the error boundary at `ui/src/shared/components/error-boundary.tsx`. Could you explain why this is separate?